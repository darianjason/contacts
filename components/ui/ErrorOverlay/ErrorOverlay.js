// @flow

import * as React from 'react';
import {View} from 'react-native';

import {Colors} from '../../../constants';
import DefaultText from '../DefaultText/DefaultText';
import Button from '../Button/Button';
import styles from './ErrorOverlay.styles';

type Props = {
  message: string,
  onConfirm: () => void,
};

const ErrorOverlay = ({message, onConfirm}: Props): React.Node => (
  <View style={styles.container}>
    <DefaultText style={styles.heading}>An error occurred</DefaultText>
    <DefaultText>{message}</DefaultText>
    <Button
      onPress={onConfirm}
      icon="redo-alt"
      size={14}
      color={Colors.accent}
      backgroundColor={Colors.primary}
      style={styles.button}
    >
      Retry
    </Button>
  </View>
);

export default ErrorOverlay;
