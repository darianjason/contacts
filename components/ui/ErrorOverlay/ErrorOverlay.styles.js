// @flow

import {StyleSheet} from 'react-native';
import type {
  ViewStyleProp,
  TextStyleProp,
} from 'react-native/Libraries/StyleSheet/StyleSheet';

import {Colors} from '../../../constants';

type Styles = {
  container: ViewStyleProp,
  heading: TextStyleProp,
  button: ViewStyleProp,
};

const styles: Styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 36,
    justifyContent: 'center',
    alignItems: 'center',
  },
  heading: {
    fontFamily: 'Manrope-Bold',
    fontSize: 18,
    marginBottom: 8,
    color: Colors.primary,
  },
  button: {
    marginTop: 24,
    alignSelf: 'center',
  },
});

export default styles;
