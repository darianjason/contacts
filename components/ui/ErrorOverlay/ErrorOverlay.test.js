import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import renderer from 'react-test-renderer';

import ErrorOverlay from './ErrorOverlay';

describe('Error Overlay component', () => {
  test('renders error message', () => {
    const errorMessage = 'Error message';

    const {getByText} = render(<ErrorOverlay message={errorMessage} />);

    expect(getByText('Error message')).toBeTruthy();
  });

  test('retry button can be pressed', () => {
    const onConfirm = jest.fn();

    const {getByText} = render(<ErrorOverlay onConfirm={onConfirm} />);

    fireEvent.press(getByText('Retry'));
    expect(onConfirm).toHaveReturnedTimes(1);
  });

  test('renders correctly', () => {
    const errorMessage = 'Error message';
    const onConfirm = jest.fn();

    const tree = renderer
      .create(<ErrorOverlay message={errorMessage} onConfirm={onConfirm} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
