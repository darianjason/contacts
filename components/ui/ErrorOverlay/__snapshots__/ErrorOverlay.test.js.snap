// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Overlay component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "padding": 36,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#142B3E",
        "fontFamily": "Manrope-Bold",
        "fontSize": 18,
        "marginBottom": 8,
      }
    }
  >
    An error occurred
  </Text>
  <Text
    style={
      Object {
        "color": "#060A0E",
        "fontFamily": "Manrope-Regular",
      }
    }
  >
    Error message
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
        false,
        Object {
          "borderRadius": 32,
          "flexDirection": "row",
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        },
        Object {
          "backgroundColor": "#142B3E",
        },
        Object {
          "alignSelf": "center",
          "marginTop": 24,
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": "#E2EEF3",
            "fontSize": 14,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome5Free-Solid",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {
            "fontWeight": "900",
          },
        ]
      }
    >
      ï‹¹
    </Text>
    <Text
      style={
        Object {
          "color": "#E2EEF3",
          "fontFamily": "Manrope-Regular",
          "marginLeft": 8,
        }
      }
    >
      Retry
    </Text>
  </View>
</View>
`;
